device/core/test/_module_client_test.js

```diff
@@ -5,12 +5,43 @@
 
 var EventEmitter = require('events');
 var assert = require('chai').assert;
-var errors = require('azure-iot-common').errors;
 var sinon = require('sinon');
+var FakeTransport = require('./fake_transport.js');
+var Message = require('azure-iot-common').Message;
+var ModuleClient = require('../lib/module_client').ModuleClient;
+var errors = require('azure-iot-common').errors;
 var ModuleClient = require('../lib/module_client').ModuleClient;
 var IotEdgeAuthenticationProvider = require('../lib/iotedge_authentication_provider').IotEdgeAuthenticationProvider;
 
-describe('ModuleClient', function() {
+describe('ModuleClient', function () {
+  var sharedKeyConnectionString = 'HostName=host;DeviceId=id;ModuleId=modId;SharedAccessKey=key';
+  var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net/devices/deviceId/modules/moduleId&sig=s1gn4tur3&se=1454204843"';
+
+  describe('#fromConnectionString', function () {
+    /*Tests_SRS_NODE_INTERNAL_CLIENT_05_006: [The fromConnectionString method shall return a new instance of the InternalClient object, as by a call to new InternalClient(new Transport(...)).]*/
```

Comment:
>Tests_SRS_NODE_INTERNAL_CLIENT_05_006 [](start = 6, length = 37)

I hate renaming SRS names, but do you want to rename the SRS name?  Also, this refers to the InternalClient class.

Reply:
All comments in here that apply to one factory method apply to all factory methods. I'm not going to comment each time.

---
In reply to: [195808304](https://github.com/Azure/azure-iot-sdk-node/pull/316#discussion_r195808304) [](ancestors = 195808304)